src/analysis/normed_space/finite_dimension.lean

```diff
@@ -0,0 +1,255 @@
+  resolution. It is however registered as an instance for `𝕜 = ℝ`.
+
+## Implementation notes
+
+The fact that all norms are equivalent is not written explicitly, as it would mean having two norms
+on a single space, which is not the way type classes work. However, if one has a
+finite-dimensional vector space `E` with a norm, and a copy `E'` of this type with another norm,
+then the identities from `E` to `E'` and from `E'`to `E` are continuous thanks to
+`linear_map.continuous_of_finite_dimensional`. This gives the desired norm equivalence.
+
+The proofs rely on linear equivalences, which are only defined in mathlib for types in the same
+universe. Therefore, all the results in this file are restricted to spaces leaving in the same
+universe as their base field.
+-/
+
+universes u v
+
+open set finite_dimensional
+open_locale classical
+
+-- To get a reasonable compile time for `continuous_equiv_fun_basis`, typeclass inference needs
+-- to be guided.
+local attribute [instance, priority 10000] pi.module normed_space.to_vector_space
+  vector_space.to_module submodule.add_comm_group submodule.module
+  linear_map.finite_dimensional_range Pi.complete nondiscrete_normed_field.to_normed_field
+
+set_option class.instance_max_depth 100
+
+/-- A linear map on `ι → 𝕜` (where `ι` is a fintype) is continuous -/
+lemma linear_map.continuous_on_pi {ι : Type u} [fintype ι] {𝕜 : Type u} [normed_field 𝕜]
+  {E : Type v} [normed_group E] [normed_space 𝕜 E] (f : (ι → 𝕜) →ₗ[𝕜] E) : continuous f :=
+begin
+  -- for the proof, write `f` in the standard basis, and use that each coordinate is a continuous
+  -- function.
+  have : (f : (ι → 𝕜) → E) =
+         (λx, finset.sum finset.univ (λi:ι, x i • (f (λj, if i = j then 1 else 0)))),
+    by { ext x, exact f.pi_apply_eq_sum_univ x },
+  rw this,
+  refine continuous_finset_sum _ (λi hi, _),
+  exact continuous_smul (continuous_apply i) continuous_const
+end
+
+section complete_field
+
+-- we use linear equivs, which require all the types to live in the same universe
+variables {𝕜 : Type u} [nondiscrete_normed_field 𝕜]
+{E : Type u} [normed_group E] [normed_space 𝕜 E]
+{F : Type v} [normed_group F] [normed_space 𝕜 F]
+[complete_space 𝕜]
+
+set_option class.instance_max_depth 150
+
+/-- In finite dimension over a complete field, the canonical identification (in terms of a basis)
+with 𝕜^n together with its sup norm is continuous. This is the nontrivial part in the fact that all
+norms are equivalent in finite dimension.
+Do not use this statement as its formulation is awkward (in terms of the dimension n, as the proof
+is done by induction over n) and it is superceded by the fact that every linear map on a
+finite-dimensional space is continuous, in `linear_map.continuous_of_finite_dimensional`. -/
+lemma continuous_equiv_fun_basis {n : ℕ} {ι : Type u} [fintype ι] (ξ : ι → E)
+  (hn : fintype.card ι = n) (hξ : is_basis 𝕜 ξ) : continuous (equiv_fun_basis hξ) :=
+begin
+  resetI,
```

Comment:
Does a `resetI` at the beginning of a proof do something?

Reply:
Yes. If I remove the `resetI` and the `classical` on the next line (which shouldn't be here, since I have opened the locale), then the induction complains that `failed to revert '_inst_2', it is a frozen local instance (possible solution: use tactic `tactic.unfreeze_local_instances` to reset the set of local instances)`. Keeping either `resetI` or `classical` solves this. The thing is I generalize `E` in the induction, and `E` has several instances attached to it.

Reply:
So, shouldn't you be using `unfreeze_*` or whatever the tactic is actually called?

Reply:
You're right, the correct mathlib function for this is `unfreezeI`, although `resetI` also does the job. I have made the change.